pkg/manager/manager.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2019 The KubeMacPool Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package manager
+
+import (
+	"flag"
+	"fmt"
+	"k8s.io/client-go/rest"
+	"net"
+	"os"
+	"os/signal"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"time"
+
+	"github.com/go-logr/logr"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
+	"github.com/K8sNetworkPlumbingWG/kubemacpool/pkg/controller"
+	poolmanager "github.com/K8sNetworkPlumbingWG/kubemacpool/pkg/pool-manager"
+	"github.com/K8sNetworkPlumbingWG/kubemacpool/pkg/webhook"
+)
+
+var log logr.Logger
+
+type KubeMacPoolManager struct {
+	clientset                *kubernetes.Clientset
+	config                   *rest.Config
+	metricsAddr              string
+	continueToRunManager     bool
+	restartChannel           chan struct{}
+	kubevirtInstalledChannel chan struct{}
+	stopSignalChannel        chan os.Signal
+}
+
+func NewKubeMacPoolManager() *KubeMacPoolManager {
+	kubemacpoolManager := &KubeMacPoolManager{continueToRunManager: true,
+		restartChannel:           make(chan struct{}),
+		kubevirtInstalledChannel: make(chan struct{}),
+		stopSignalChannel:        make(chan os.Signal, 1)}
+
+	signal.Notify(kubemacpoolManager.stopSignalChannel, os.Interrupt, os.Kill)
+
+	flag.StringVar(&kubemacpoolManager.metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
+	var logType string
+	flag.StringVar(&logType, "v", "production", "Log type (debug/production).")
+	flag.Parse()
+
+	if logType == "debug" {
+		logf.SetLogger(logf.ZapLogger(true))
+	} else {
+		logf.SetLogger(logf.ZapLogger(false))
+	}
+	log = logf.Log.WithName("manager")
+
+	return kubemacpoolManager
+}
+
+func (k *KubeMacPoolManager) Run() {
+	// setup Pool Manager
+	rangeStart := loadMacAddressFromEnvVar(poolmanager.RangeStartEnv)
```

Comment:
With env vars, I would be happier if they were loaded in main, but that is sometimes annoying once you need to pass the value all the way to its user, so I won't fight for this one.